updated and expanded documentation (#12)
simplyanalytics/client-lib-python: docs/data_retrieval.md

| `geographicRelease`             | The geographic release of a location. This represents updates within a census.                                                                                                          |
| `country`                       | The country of a location.                                                                                                                                                              |
| `extent`                        | The geographic extent (i.e. bounding box) of a location.                                                                                                                                |
| `polygon`                       | The polygon of a location.                                                                                                                                                              |
| `polygon->{geographic release}` | Polygon attributes for specific annual geographic releases, e.g. `polygon->2020`                                                                                                        |

> Additionally, there are attributes for each geographic unit for each location representing parent relationships. For example, a block group has a `usa:censusTract` attribute that indicates its parent census tract.

Reviewer: Please feel free to ignore, but I would phrase it like this to remove the second "for":
```suggestion
Additionally, each location has an attribute for each geographic unit representing parent relationships. For example, a block group has a `usa:censusTract` attribute that indicates its parent census tract.
```

Also, is the value of a parent relationship attribute `null` when the location's geographic unit is bigger than the geographic unit after which that parent relationship attribute is named? Should we specify that behaviour?

> Note that some locations have multiple parents of the same unit. For example, it's possible for a ZIP code to be in two places at once.

Reviewer: Should we specify how such a parent relationship attribute's value would be returned?

Author: That's my preference as well, but as mentioned, I'd prefer not changing it until we have time to test it more.

Attribute arguments are represented as objects with an `attribute` key set to either an attribute ID or attribute series ID. When set to an attribute ID (e.g. `USACSSUB->income_1->2025.2`) the exact attribute is used. When set to an attribute series ID (e.g. `USACSSUB->income_1`) the latest available attribute in that series is used.

### Comparison
- `=`, `<>`, `>`, `>=`, `<`, `<=`
    - Standard comparison operators that compare an attribute or the result of a function with a literal value.
    - Arguments: Attribute or function, string or number.
    - Example: Select all US states.
      ```json
      ["=", {"attribute": "geographicUnit"}, "usa:state"]
      ```
- `~`
    - A fuzzy text match that matches if the attribute contains the words in the string after first converting both to normalized lexemes.
    - Arguments: a valid attribute and search string
        - *This is only available for two business point attributes: `names` and `code_descriptions`*.
    - Example: Search D&B dataset for businesses in New York state containing "working horse" in any of their names.
      ```json
      [
          "and",
          ["within", "usa:state:36"],

>           ["~", {"attribute": "DBUSA->names"}, "working horse"]

Reviewer 2: Should probably change this if business data isn't going to be available

              [">", {"attribute": "USACSSUB->income_1"}, 200000],
              [">", {"attribute": "USACSSUB->education_20_pct"}, 90]
          ]
      ]
      ```
- `not`
    - Inverts the result of the argument: true if false or false if true.
    - Arguments: One operation.
    - Example: Select a "donut" of block groups around New York City by selecting block groups within 10 miles, and not those within 5 miles.
      ```json
      [
            "and",
            ["radius", "usa:censusBlockGroup", 16093.44, "usa:place:3651000"],
            ["not", ["radius", "usa:censusBlockGroup", 8046.72, "usa:place:3651000"]]
      ]
      ```

## System Attributes
The following system attributes are available to all users in addition to subscription data.

| Attribute ID                    | Description                                                                                                                                                                      |
|---------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `locationSeries`                | An ID that uniquely identifies a location regardless of time. This is similar to a "spatial ID" but globally unique. For example, `usa:state:04` always refers to Arizona state. |

Reviewer 2: ```suggestion
| `locationSeries`                | An ID that uniquely identifies a location regardless of time. This is similar to a "spatial ID" but globally unique. For example, `usa:state:04` always refers to the state of Arizona. |
```

| `polygon`                       | The polygon of a location.                                                                                                                                                       |
| `polygon->{geographic release}` | Polygon attributes for specific annual geographic releases, e.g. `polygon->2020`                                                                                                 |

Additionally, there are attributes for each geographic unit for each location representing parent relationships. For example, a block group has a `usa:censusTract` attribute that indicates its parent census tract.

Note that some locations have multiple parents of the same unit. For example, it's possible for a ZIP code to be in two places at once.

### Geographic Units

SimplyAnalytics has data for the following geographic units. Not all datasets have data for all units. The actual units available depend on your data subscription and the units available for your datasets.

### USA:

> | Geographic Unit ID           | Name                              |

Reviewer 2: same with `usa:duns/can:duns` again.
